sorc/build_all.sh

```diff
@@ -82,20 +64,94 @@ while getopts ":a:dfghj:kuvwy" option; do
       ;;
   esac
 done
-
 shift $((OPTIND-1))
 
+# If no build system was specified, build for gfs forecast-only
+if [[ $# -eq 0 ]]; then
+   selected_systems="gfs"
+else
+   selected_systems="$*"
+fi
+
+supported_systems=("gfs" "gefs" "sfs" "gsi" "gdas" "all")
+# shellcheck disable=SC2034
+gfs_builds="gfs gfs_utils ufs_utils upp ww3_gfs"
+# shellcheck disable=SC2034
+gefs_builds="gefs gfs_utils ufs_utils upp ww3_gefs"
+# shellcheck disable=SC2034
+sfs_builds="sfs gfs_utils ufs_utils upp ww3_gefs"
+# shellcheck disable=SC2034
+gsi_builds="gsi_enkf gsi_monitor gsi_utils"
+# shellcheck disable=SC2034
+gdas_builds="gdas gsi_monitor gsi_utils"
+# shellcheck disable=SC2034
+all_builds="gfs gfs_utils ufs_utils upp ww3_gfs gefs ww3_gefs gdas gsi_enkf gsi_monitor gsi_utils"
+
 logs_dir="${HOMEgfs}/sorc/logs"
 if [[ ! -d "${logs_dir}" ]]; then
   echo "Creating logs folder"
   mkdir -p "${logs_dir}" || exit 1
 fi
 
-# Check final exec folder exists
-if [[ ! -d "${HOMEgfs}/exec" ]]; then
-  echo "Creating ${HOMEgfs}/exec folder"
-  mkdir -p "${HOMEgfs}/exec"
-fi
+# Jobs per build ("min max")
+declare -A build_jobs build_opts build_scripts
+build_jobs=(
+    ["gfs"]=8 ["gefs"]=8 ["sfs"]=8 ["gdas"]=8 ["gsi_enkf"]=2 ["gfs_utils"]=1 ["ufs_utils"]=1
+    ["ww3_gfs"]=1 ["ww3_gefs"]=1 ["gsi_utils"]=1 ["gsi_monitor"]=1 ["gfs_utils"]=1 ["upp"]=1
+)
+
+# Establish build options for each job
+_gfs_exec="gfs_model.x"
+_gefs_exec="gefs_model.x"
+_sfs_exec="sfs_model.x"
+build_opts=(
```

Comment:
Might be clearer to not separate the build script from the build options and just have one dict.

Reply:
This makes most things easier, but logging errors is a little tricky here:

https://github.com/NOAA-EMC/global-workflow/blob/72c8d7ac65a4f260aafad302c4ce50a45a1b1253/sorc/build_all.sh#L286-L290

I think I will save this for a later rendition.